Library/Homebrew/test/test_utils.rb

```diff
@@ -206,4 +206,13 @@ def test_number_readable
     assert_equal "1,000", number_readable(1_000)
     assert_equal "1,000,000", number_readable(1_000_000)
   end
+
+  def test_user_agent
+    ENV.delete "CI"
+    ENV.delete "JENKINS_URL"
+    refute_includes homebrew_user_agent, "CI"
+    ENV["CI"] = "true"
+    assert_includes homebrew_user_agent, "CI"
+    assert_includes homebrew_user_agent, HOMEBREW_VERSION
```

Comment:
I suggest to back up ENV here, like below, to avoid pollute ENV for other tests.
```ruby
def test_user_agent
  old_env = ENV.dup
  ...
ensure
  ENV = old_env
end
```